.github/workflows/pr-auto-assign.yml

```diff
@@ -16,11 +16,21 @@ jobs:
           script: |
             const reviewers = ['mstoykov', 'imiric', 'codebien', 'olegbespalov', 'oleiade'];
             const reviewerCount = 2;
+            const crypto = require("node:crypto");
+
+            const getNRandom = (n, array) => {
+              let result = new Array();
+              for (;n > 0 && array.length > 0; n--) {
+                const chosen = array[crypto.randomInt(array.length)];
+                result.push(chosen);
+                array = array.filter(el => el != chosen);
+              }
+              return result;
+            }
 
-            const reviewersWithoutAuthor = reviewers.filter(user => user !== context.payload.pull_request.user.login);
             github.rest.pulls.requestReviewers({
               owner: context.repo.owner,
               repo: context.repo.repo,
               pull_number: context.payload.pull_request.number,
-              reviewers: reviewersWithoutAuthor.sort(() => 0.5 - Math.random()).slice(0, reviewerCount),
+              reviewers: getNRandom(reviewerCount, reviewers),
```

Comment:
Interesting. I wasn't aware that using `Array.sort()` for random shuffling would be biased towards certain elements, depending on the underlying implementation.

I ran some tests and confirmed it:

```javascript
// auto-assign-test.js
const assignees = ['mstoykov', 'imiric', 'codebien', 'olegbespalov', 'oleiade'];
const assigneeCount = 1;
console.log(assignees.sort(() => 0.5 - Math.random()).slice(0, assigneeCount)[0]);
```
```shell
$ for i in {0..1000}; do node auto-assign-test.js; done > auto-assign-test-results.txt
$ sort auto-assign-test-results.txt | uniq -c | sort
    120 imiric
    155 codebien
    165 oleiade
    229 olegbespalov
    332 mstoykov
```

:smile:

This is explained in more detail [here](https://stackoverflow.com/a/962829) and [here](https://javascript.info/array-methods#shuffle-an-array) (click on "solution").

Reply:
Well, I guess this shows what copy-pasting a random StackOverflow answer after a cursory "that seems about right" check will get you  :sweat_smile:  Sorry about that! :bow: 

And now, with ChatGPT/Co-pilot/etc, the future is looking very bright indeed :sob: